Exercises, Evaluations, and Printable Worksheet scripts added to each page.

qsebring3930 opened the pull request:
-exercises.js added to the default scripts that a .md page loads with
-exercises.js adds the exercises portions, evaluation portions, and printable worksheet portions to a page
-exercises and evaluations each pull data from a .txt file that holds file source destinations, questions, and answers
-printable worksheet portion toggles based on the available questions, sends a token to a new page "worksheet.md" that is only directed to from the print function.
-worksheet.md has a script that builds the proper worksheet from the token information.

qsebring3930 committed:
exercises.js created for default page template

each page now has an exercises portion

qsebring3930 committed:
evaluations added to exercises.js page template

qsebring3930 committed:
printing added to exercises.js template

qsebring3930 committed:
styled exercises/evaluation/print template to match rest of page

qsebring3930 committed:
worksheet.md added and printables now formatted.

mgoadric commented:
This looks great! Since it shows up on every page, and only has the test questions, I won't merge it yet. We should make it on/off for each page, maybe with some flag in the frontmatter portion of the page, so we can keep the website functional and start incorporating this bit by bit.

qsebring3930 committed:
Evaluation, Exercises, and Printables are only added to page when their respective files exist.

qsebring3930 commented:
I've added a file exists bool at the beginning of the script that checks if the respective files exist, it either returns a 404 to console or continues with populating the page. I did forget to change the script to pull information from the respective page file, so I added that also. You can check on Gnomesort to see that only its exercise portion is populated since it is the only page that has an exercises file.

qsebring3930 committed:
Added 2-3 Trees video, algorithm, example

mgoadric commented:
Perfect! Nice job with the 2-3 tree too!
